
BYON throws IndexOutOfBoundsException if
the #of available nodes is 0 and the cluster wants 1+

running InitScript{INSTANCE_NAME=bootstrap-hadoop-datanode_hadoop-tasktracker} on node(hdp1)
Not enough nodes available for template hadoop-namenode+hadoop-jobtracker
*CRITICAL* the cluster failed to launch and the automated node termination option was not selected, there might be orphaned nodes.
java.lang.IndexOutOfBoundsException: toIndex = 1
	at java.util.SubList.<init>(AbstractList.java:602)
	at java.util.RandomAccessSubList.<init>(AbstractList.java:758)
	at java.util.AbstractList.subList(AbstractList.java:468)
	at org.apache.whirr.actions.ByonClusterAction.doAction(ByonClusterAction.java:137)
	at org.apache.whirr.actions.ScriptBasedClusterAction.execute(ScriptBasedClusterAction.java:126)
	at org.apache.whirr.ByonClusterController.bootstrapCluster(ByonClusterController.java:83)
	at org.apache.whirr.ClusterController.launchCluster(ClusterController.java:113)
	at org.apache.whirr.cli.command.LaunchClusterCommand.run(LaunchClusterCommand.java:69)
	at org.apache.whirr.cli.command.LaunchClusterCommand.run(LaunchClusterCommand.java:59)
	at org.apache.whirr.cli.Main.run(Main.java:69)
	at org.apache.whirr.cli.Main.main(Main.java:102)
Exception in thread "main" java.lang.RuntimeException: java.lang.IndexOutOfBoundsException: toIndex = 1
	at org.apache.whirr.ClusterController.launchCluster(ClusterController.java:128)
	at org.apache.whirr.cli.command.LaunchClusterCommand.run(LaunchClusterCommand.java:69)
	at org.apache.whirr.cli.command.LaunchClusterCommand.run(LaunchClusterCommand.java:59)
	at org.apache.whirr.cli.Main.run(Main.java:69)
	at org.apache.whirr.cli.Main.main(Main.java:102)
Caused by: java.lang.IndexOutOfBoundsException: toIndex = 1
	at java.util.SubList.<init>(AbstractList.java:602)
	at java.util.RandomAccessSubList.<init>(AbstractList.java:758)
	at java.util.AbstractList.subList(AbstractList.java:468)
	at org.apache.whirr.actions.ByonClusterAction.doAction(ByonClusterAction.java:137)
	at org.apache.whirr.actions.ScriptBasedClusterAction.execute(ScriptBasedClusterAction.java:126)
	at org.apache.whirr.ByonClusterController.bootstrapCluster(ByonClusterController.java:83)
	at org.apache.whirr.ClusterController.launchCluster(ClusterController.java:113)
	... 4 more


Failure of the JT to start up due to a bad user name doesn't get picked up:

+ break
+ return 0
+ echo 'Starting hadoop-jobtracker'
+ service hadoop-jobtracker start
chown: invalid user: `hadoop'
/usr/lib/hadoop/bin/hadoop-daemon.sh: line 135: /var/log/hadoop/logs/hadoop-hadoop-jobtracker-nn1.hortonworks.com.out: Permission denied
head: cannot open `/var/log/hadoop/logs/hadoop-hadoop-jobtracker-nn1.hortonworks.com.out' for reading: No such file or directory
+ retval=0
+ (( 0 == 0 ))
+ echo 'Service hadoop-jobtracker is started'
+ CONFIGURE_HADOOP_DONE=1
+ cat
+ cat
+ cat
+ cat
